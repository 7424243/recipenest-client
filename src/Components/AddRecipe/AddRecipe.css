h3 {
    padding-top: 20px;
    text-align: center;
}

input {
    margin-bottom: 10px;
}

h3 {
    font-family: 'Baskervville', serif;
    font-size: 25px;
}

input, label, textarea {
    display: block;
    width: 100%;
    font-family: 'Baskervville', serif;
    font-size: large;
}

label {
    margin-top: 10px;
    margin-bottom: 5px;
}


/* Classes */

.add-recipe-form {
    margin: 0 auto;
    padding: 0 0 40px;
    max-width: 80%;
    text-align: left;
}

.buttons {
    margin-top: 40px;
    text-align: center;
    font-family: 'Baskervville', serif;
}


/* @media */

@media screen and (min-width: 540px) {
    .add-recipe-form {
        padding: 0 0 40px;
        max-width: 450px;
    }
}