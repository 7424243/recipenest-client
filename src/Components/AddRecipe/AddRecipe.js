import React, {Component} from 'react'
import { Link } from 'react-router-dom'
import RecipenestContext from '../../RecipenestContext'
import './AddRecipe.css'

class AddRecipe extends Component {

    static contextType = RecipenestContext

    render() {
        return (
            <>
                <h3>New Recipe</h3>
                <form className='add-recipe-form'>
                    <section className="name">
                        <label htmlFor="name">Recipe Name: </label>
                        <input type="text" placeholder="Recipe Name"/>
                    </section>
                        <section className="url">
                        <label htmlFor="url">Recipe URL: </label>
                        <input type="url" placeholder="https://www.recipe.com" />
                    </section>
                    <section className="like">
                        <label htmlFor="like">Description: </label>
                        <textarea name="like" id="" cols="30" rows="10" placeholder="The word cloud will be make out of this input"></textarea>
                    </section>
                    <section className="notes">
                        <label htmlFor="notes">Additional Notes: </label>
                        <textarea name="notes" id="" cols="30" rows="10" placeholder="Modifications for next time? Likes? Dislikes?"></textarea>
                    </section>
                    <section className="buttons">
                        <Link to="/recipes"><button>Cancel</button></Link>
                        <button>Save</button>
                    </section>
                </form>
            </>
        )
    }
}

export default AddRecipe