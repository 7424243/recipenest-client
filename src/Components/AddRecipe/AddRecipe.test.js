import React from 'react'
import ReactDOM from 'react-dom'
import {BrowserRouter} from 'react-router-dom'
import renderer from 'react-test-renderer'
import AddRecipe from './AddRecipe';

describe('AddRecipe Component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<BrowserRouter><AddRecipe/></BrowserRouter>, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<BrowserRouter><AddRecipe/></BrowserRouter>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})

