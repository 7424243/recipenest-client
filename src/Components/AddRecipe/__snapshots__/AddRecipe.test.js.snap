// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddRecipe Component renders the UI as expected 1`] = `
<div
  className="add-recipe-container"
>
  <h3>
    New Recipe
  </h3>
  <form
    className="add-recipe-form"
    onSubmit={[Function]}
  >
    <section
      className="name"
    >
      <label
        htmlFor="name"
      >
        Recipe Name: 
      </label>
      <input
        aria-label="recipe name"
        name="name"
        onChange={[Function]}
        placeholder="recipe name"
        required={true}
        type="text"
      />
    </section>
    <section
      className="url"
    >
      <label
        htmlFor="url"
      >
        Recipe URL: 
      </label>
      <input
        aria-label="recipe url"
        name="url"
        onChange={[Function]}
        placeholder="https://www.recipe.com"
        required={true}
        type="url"
      />
    </section>
    <section
      className="notes"
    >
      <label
        htmlFor="notes"
      >
        Notes: 
      </label>
      <textarea
        aria-label="Modifications for next time? Likes? Dislikes?"
        cols="30"
        name="notes"
        onChange={[Function]}
        placeholder="Modifications for next time? Likes? Dislikes?"
        required={true}
        rows="5"
      />
    </section>
    <section
      className="url"
    >
      <label
        htmlFor="img-url"
      >
        Image URL: (or leave blank to use a default image)
      </label>
      <input
        aria-label="add image url or leave blank to use a default image"
        name="img-url"
        onChange={[Function]}
        placeholder="https://www.recipe.com"
        type="url"
      />
    </section>
    <section
      className="buttons"
    >
      <a
        href="/my-recipes"
        onClick={[Function]}
      >
        <button>
          Cancel
        </button>
      </a>
      <button
        type="submit"
      >
        Save
      </button>
    </section>
  </form>
</div>
`;
